cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/ParameterUtils.java

```diff
@@ -177,7 +177,11 @@ public static CruiseControlEndPoint endPoint(HttpServletRequest request) {
         throw new UserRequestException("Unsupported request method: " + request.getMethod() + ".");
     }
 
-    String path = request.getRequestURI().toUpperCase().replace(REQUEST_URI, "");
+    if (apiUrlPrefix.endsWith("/*")) {
```

Comment:
`apiUrlPrefix` must end with `/*` -- i.e. failure of this condition must result in a configuration time failure.
Hence, this if check is redundant.